traffic_portal/app/src/common/api/DeliveryServiceSslKeysService.js

```diff
@@ -17,75 +17,80 @@
  * under the License.
  */
 
-var DeliveryServiceSslKeysService = function($http, $q, locationUtils, messageModel, ENV) {
+var DeliveryServiceSslKeysService = function($http, locationUtils, messageModel, ENV) {
+    this.successMessage = 'SSL Keys generated and updated for ';
+    this.letsEncryptSuccessMessage = 'Call to Lets Encrypt has been made successfully. This may take a few minutes. Please watch for a notification in the Change Log. Delivery Service = ';
+
 	this.generateSslKeys = function(deliveryService, sslKeys, generateSslKeyForm) {
-		if (sslKeys.hasOwnProperty('version')){
-			generateSslKeyForm.version = parseInt(sslKeys.version) + 1;
-		} else {
-			generateSslKeyForm.version = 1;
-		}
+		 return this.generateSslKeysBase(deliveryService, sslKeys, generateSslKeyForm, 'deliveryservices/sslkeys/generate', this.successMessage);
+	};
 
-		generateSslKeyForm.cdn = deliveryService.cdnName;
-		generateSslKeyForm.deliveryservice = deliveryService.xmlId;
-		generateSslKeyForm.key = deliveryService.xmlId;
+    this.generateSslKeysWithLetsEncrypt = function(deliveryService, sslKeys, generateSslKeyForm) {
+        return this.generateSslKeysBase(deliveryService, sslKeys, generateSslKeyForm, 'deliveryservices/sslkeys/generate/letsencrypt', this.letsEncryptSuccessMessage);
+    };
 
-		var request = $q.defer();
-        $http.post(ENV.api['root'] + "deliveryservices/sslkeys/generate", generateSslKeyForm)
-        .then(
-            function(result) {
-            	messageModel.setMessages([ { level: 'success', text: 'SSL Keys generated and updated for ' + deliveryService.xmlId } ], true);
-                request.resolve(result.data.response);
-            },
-            function(fault) {
-            	messageModel.setMessages(fault.data.alerts, false);
-                request.reject(fault);
-            }
-        );
-        return request.promise;
-	};
+	this.generateSslKeysBase = function(deliveryService, sslKeys, generateSslKeyForm, endpoint, message) {
+        if (sslKeys.hasOwnProperty('version')){
+            generateSslKeyForm.version = parseInt(sslKeys.version, 10) + 1;
+        } else {
+            generateSslKeyForm.version = 1;
+        }
+
+        generateSslKeyForm.cdn = deliveryService.cdnName;
+        generateSslKeyForm.deliveryservice = deliveryService.xmlId;
+        generateSslKeyForm.key = deliveryService.xmlId;
+
+        return $http.post(ENV.api['root'] + endpoint, generateSslKeyForm)
+            .then(
+                function(result) {
+                    messageModel.setMessages([ { level: 'success', text: message + deliveryService.xmlId } ], true);
```

Comment:
It looks like here you could just pass in the `alerts` from the server response, couldn't you? I think you've re-implemented what it already does.

Also, if you're not redirecting after setting the messages, that second argument should be `false`.

Reply:
i did it this way so that i could change which message it shows based on if it called LE or not.  and it does redirect after